### ISSUE TITLE
User Importer doesn't check if a user exists but is marked deleted before creating a new one.

### ISSUE THREAD

**Issue by loganhawker** [grokability/snipe-it#9292]
#### Please confirm you have done the following before posting your bug report:

- :white_check_mark: I have enabled debug mode 
- :white_check_mark: I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

**Describe the bug**
User Importer doesn't check if a user exists but is marked deleted before creating a new one.

**To Reproduce**
1. Use snipeit:import to import a user from CSV
2. Delete the user
3. Re-import the CSV
4. See that a new user is created with a new id, and that there is the same user under Show Deleted & Show Current

**Expected behavior**
Importing the CSV again would restore the user if they were marked as deleted.

**Server (please complete the following information):**
 - Snipe-IT Version [ v5.1.2 - build 5847 (master) ]
 - OS: [Windows Server 2019]
 - Web Server: [Apache]
 - PHP Version [7.4.9]

**Desktop (please complete the following information):**
 - OS: [ChromeOS]
 - Browser [Chrome]
 - Version [89]

**Error Messages**
- WITH DEBUG TURNED ON, if you're getting an error in your browser, include that error
   Using command line
- If a stacktrace is provided in the error, include that too.
   Using command line
- Any errors that appear in your browser's error console.
   Using command line
- Confirm whether the error is reproducible on the demo: https://snipeitapp.com/demo.
   Using command line
- Include any additional information you can find in `storage/logs` and your webserver's logs.
   My logs only have a laravel.txt with stacktraces from me testing LDAP 
- Include the output from `php -m` (this should display what modules you have enabled.)
   https://pastebin.com/PexRck4R

**Additional context**
- Fresh Install
- Windows Server 2019
- Wamp Server 3.2.3
- Install.sh
- Include what you've done so far in the installation, and if you got any error messages along the way.
     Imported users with snipeit:import 
- Indicate whether or not you've manually edited any data directly in the database
     No

Please do not post an issue without answering the related questions above. If you have opened a different issue and already answered these questions, answer them again, once for every ticket. It will be next to impossible for us to help you.

**Comment by snipe** [comment]
I don't think that really makes sense though. If you have a Jane Smith, she leaves the company, and then you hire another Jane Smith, we wouldn't want the second one to inherit that previous user's history.
